Library/Homebrew/formulary.rb

```diff
@@ -34,43 +34,64 @@ module Formulary
   # @api internal
   sig { void }
   def self.enable_factory_cache!
-    @factory_cache = true
+    @factory_cache_enabled = T.let(true, T.nilable(T::Boolean))
```

Comment:
Based on my reading of line 44, this would behave erroneously if set to false:
```suggestion
    @factory_cache_enabled = T.let(true, T.nilable(T::TrueClass))
```
(Even with the above, It may still be a good idea to defensively update `factory_cached?` to handle a `false` value as expected)